python/lsst/ts/ess/common/socket_server.py

```diff
@@ -147,15 +149,19 @@ async def read_loop(self) -> None:
                         assert self.command_handler is not None
                         await self.command_handler.handle_command(cmd, **kwargs)
 
-                except asyncio.IncompleteReadError:
-                    self.log.exception("Read error encountered. Retrying.")
-
-        except asyncio.CancelledError:
-            self.log.debug("read_loop cancelled.")
         except Exception:
-            self.log.exception("read_loop failed.")
+            self.log.exception("read_loop failed. Disconnecting.")
+            self.log.debug("Stoppping sending telemetry.")
+            assert self.command_handler is not None
+            await self.command_handler.stop_sending_telemetry()
+            await self.disconnect()
 
     async def disconnect(self) -> None:
+        """Stop sending telemetry and close the client."""
+        self.log.debug("Cancelling read_loop_task.")
+        self.read_loop_task.cancel()
+        self.read_loop_task = asyncio.Future()
```

Comment:
Is this line necessary? I hope you can delete it. It has the confusing effect that `read_loop_task.done()` will be false when the loop is *not* running. Whereas without the line `read_loop_task.done()` is done if and only if the task is not running (assuming you don't have similar code anywhere else).

Reply:
There was a line that sets `read_loop_task` to done before creating a new one and I removed that and this line.